### PR TITLE
Add comments to customer logic

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | I added comments to Customer, Context and CustomerPersister classes here and there. There are some lines moved a bit and removed a big if condition.
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| Related PRs       | 
| How to test?      | 
| Possible impacts? | no

### CODE REVIEW

File: classes/Customer.php [hunk 1]
```diff
@@ -1193,61 +1207,80 @@ public function transformToCustomer($idLang, $password = null)
...
-                (int) $this->id_shop
-            );
 
-            return true;
+        if (!$this->update()) {
+            return false;
         }
 
-        return false;
+        $language = new Language($idLang);
```
Comment (nicosomb) [inline classes/Customer.php:1230]: Why did you move this block further?

File: classes/Customer.php [hunk 2]
```diff
@@ -1193,61 +1207,80 @@ public function transformToCustomer($idLang, $password = null)
...
-                null,
-                null,
-                null,
-                _PS_MAIL_DIR_,
-                false,
-                (int) $this->id_shop
-            );
 
-            return true;
+        if (!$this->update()) {
```
Comment (nicosomb) [inline classes/Customer.php:1226]: I like this kind of change (you exit the method sooner)

File: classes/Customer.php (hunk 1 above)
Comment (Hlavtox) [reply classes/Customer.php:1230]: Because it's only used for sending the email. So
A) For it to be closer to the logic where it's used.
B) It's after the early return, so it's processed only when needed.

File: classes/Customer.php (hunk 2 above)
Comment (Hlavtox) [reply classes/Customer.php:1226]: @nicosomb 🚀🚀🚀

**Review by nicosomb — DISMISSED** [review]

**Review by marsaldev — DISMISSED** [review]

File: classes/Customer.php [hunk 3]
```diff
@@ -1193,61 +1207,80 @@ public function transformToCustomer($idLang, $password = null)
...
+        if (!Validate::isLoadedObject($language)) {
+            $language = Context::getContext()->language;
+        }
+
+        /*
+        * Now, we will send out an email where he can set his new password.
+        *
+        * TODO:
+        * This 'guest_to_customer' email should be sent only if password was not provided. Otherwise,
+        * it should send an 'account' email without an URL. This is what CustomerPersister does. (These functions
```
Comment (marsaldev) [inline classes/Customer.php:1240]:
```suggestion
        * it should send an 'account' email without an URL. This is what CustomerPersister does. (These methods
```

Comment (Hlavtox) [reply classes/Customer.php:1240]: @marsaldev On every PR 😄😄😄

**Review by marsaldev — DISMISSED** [review]

**Review by nicosomb — DISMISSED** [review]

**Comment by kpodemski** [comment]
@PrestaShop/qa-team automated tests should be enough, only comments in the code + minor changes related to code

**Review by MhiriFaten — COMMENTED** [review]
Hello @kpodemski 

I see an error "Code owner review required" which means that no one of @PrestaShop/prestashop-maintainers has approved this PR.
Please check with the maintainers before putting the 'waiting for QA' label to test it :pray: 

Thanks in advance!

**Review by kpodemski — DISMISSED** [review]

**Review by MhiriFaten — DISMISSED** [review]
Hello @Hlavtox ,

The automated tests are OK :heavy_check_mark: 

QA approved


Thank you !

**Review by MhiriFaten — COMMENTED** [review]
@Hlavtox while this branch has conflicts that must to be resolved, I will remove the `QA approved` label
Ping @PrestaShop/prestashop-maintainers can you resolve it please !

Thank you.

**Comment by Hlavtox** [comment]
@MhiriFaten @kpodemski @nicosomb Rebased

**Review by nicosomb — APPROVED** [review]

**Review by marsaldev — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by MhiriFaten — APPROVED** [review]
Hello @Hlavtox ,

Automated tests are OK
QA approved :heavy_check_mark: well done!

Thank you.

**Comment by matks** [comment]
Thank you @Hlavtox
